apps/ledger-live-desktop/src/newArch/features/ModularDrawer/Web3AppWebview/AssetAndAccountDrawer.ts

```diff
@@ -0,0 +1,64 @@
+import { Account, AccountLike } from "@ledgerhq/types-live";
+import { Observable } from "rxjs";
+import { WalletAPIAccount } from "@ledgerhq/live-common/wallet-api/types";
+import { CryptoOrTokenCurrency } from "@ledgerhq/types-cryptoassets";
+import { setDrawer } from "~/renderer/drawers/Provider";
+import { SelectAccountFlow } from "../components/SelectAccountFlow";
+
+type Result = {
+  account: AccountLike;
+  parentAccount?: Account;
+};
+
+type DrawerParams = {
+  assetIds?: string[];
+  currencies?: CryptoOrTokenCurrency[];
+  accounts$?: Observable<WalletAPIAccount[]>;
+  includeTokens?: boolean;
+  onSuccess?: (account: AccountLike, parentAccount?: Account) => void;
+  onCancel?: () => void;
+};
+
+function openAssetAndAccountDrawer(params: DrawerParams): void {
+  const { assetIds, currencies, accounts$, includeTokens } = params;
+
+  const handleSuccess = (result: Result): void => {
+    setDrawer();
+    params.onSuccess?.(result.account, result.parentAccount);
+  };
+
+  const handleCancel = (): void => {
+    setDrawer();
+    params.onCancel?.();
+  };
+
+  return setDrawer(
+    SelectAccountFlow,
+    {
+      assetIds,
+      currenciesArray: currencies,
+      includeTokens,
+      accounts$,
+      onAccountSelected: (account, parentAccount) => {
+        handleSuccess({ account, parentAccount });
+      },
+    },
+    {
+      onRequestClose: handleCancel,
+    },
+  );
+}
+
+function openAssetAndAccountDrawerPromise(
+  drawerParams: Omit<DrawerParams, "onSuccess" | "onCancel">,
+) {
+  return new Promise<Result>((resolve, reject) =>
+    openAssetAndAccountDrawer({
+      ...drawerParams,
+      onSuccess: (account, parentAccount) => resolve({ account, parentAccount }),
+      onCancel: () => reject(new Error("Canceled by user")),
+    }),
+  ).finally(() => setDrawer());
```

Comment:
My bad you already call `setDrawer();` in the `handleSuccess` and `handleCancel` so the finally is useless